### PR TITLE
gh-95097: support custom tasks with missing uncancel methods

### PR DESCRIPTION
**Pull request by graingert**
* Issue: gh-95097

### CODE REVIEW

File: Lib/asyncio/tasks.py [hunk 1]
```diff
@@ -67,14 +67,28 @@ def _set_task_name(task, name):
         try:
             set_name = task.set_name
         except AttributeError:
-            warnings.warn("Task.set_name() was added in Python 3.8, "
+            warnings._deprecated(
+                name="missing asyncioTask.set_name()",
```
Comment (graingert) [inline Lib/asyncio/tasks.py]:
```suggestion
                name="missing asyncio.Task.set_name()",
```

File: Lib/asyncio/tasks.py [hunk 2]
```diff
@@ -67,14 +67,28 @@ def _set_task_name(task, name):
...
+                      "task implementations since 3.13.", remove=(3, 13))
         else:
             set_name(name)
 
 
+def _uncancel(task):
+    try:
+        uncancel = task.uncancel
+    except AttributeError:
+        warnings._deprecated("Task.uncancel() was added in Python 3.11, "
```
Comment (graingert) [inline Lib/asyncio/tasks.py]:
```suggestion
        warnings._deprecated(
            name="missing asyncio.Task.uncancel()",
            message="Task.uncancel() was added in Python 3.11, "
```

File: Lib/asyncio/taskgroups.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ async def __aexit__(self, et, exc, tb):
 
         if et is not None:
             if et is exceptions.CancelledError:
-                if self._parent_cancel_requested and not self._parent_task.uncancel():
+                if self._parent_cancel_requested and not tasks._uncancel(self._parent_task.uncancel):
```
Comment (graingert) [inline Lib/asyncio/taskgroups.py]:
```suggestion
                if self._parent_cancel_requested and tasks._uncancel(self._parent_task.uncancel) == 0:
```

File: Lib/asyncio/tasks.py [hunk 4]
```diff
@@ -67,14 +67,30 @@ def _set_task_name(task, name):
         try:
             set_name = task.set_name
```
Comment (graingert) [inline Lib/asyncio/tasks.py:68]: If a task is missing set_name it's probably missing uncancel

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
Can you add a test with custom task factory ?

**Comment by kumaraditya303** [comment]
Closing as the issue was fixed by https://github.com/python/cpython/pull/95211

**Pull request closed without merging**